src/ansys/sound/core/sound_composer/track.py

```diff
@@ -0,0 +1,337 @@
+        """
+        return self.__source
+
+    @source.setter
+    def source(self, obj: AnySourceType):
+        """Set the track source."""
+        if (obj is not None) and (not (isinstance(obj, AnySourceType))):
+            raise PyAnsysSoundException(
+                "Specified source must have a valid type (SourceSpectrum, SourceBroadbandNoise, "
+                "SourceBroadbandNoiseTwoParameters, SourceHarmonics, "
+                "SourceHarmoncisTwoParameters, or SourceAudio)."
+            )
+        self.__source = obj
+
+    @property
+    def filter(self) -> Filter:
+        """Filter object of the track."""
+        return self.__filter
+
+    @filter.setter
+    def filter(self, obj: Filter):
+        """Set the track filter."""
+        if (obj is not None) and (not (isinstance(obj, Filter))):
+            raise PyAnsysSoundException("Specified filter must be of type Filter.")
+        self.__filter = obj
+
+    def process(self, sampling_frequency: float = 44100.0):
+        """Generate the signal of the track, using the current source and filter.
+
+        Parameters
+        ----------
+        sampling_frequency : float, default: 44100.0
+            Sampling frequency of the generated sound in Hz.
+        """
+        if sampling_frequency <= 0.0:
+            raise PyAnsysSoundException("Sampling frequency must be strictly positive.")
+
+        if self.source is None:
+            raise PyAnsysSoundException(f"Source is not set. Use {__class__.__name__}.source.")
+
+        self.source.process(sampling_frequency)
+        signal = self.source.get_output()
+
+        if self.filter is not None:
+            self.filter.signal = signal
+            self.filter.process()
+            signal = self.filter.get_output()
+
+        if self.gain != 0.0:
+            gain_obj = ApplyGain(signal=signal, gain=self.gain, gain_in_db=True)
+            gain_obj.process()
+            signal = gain_obj.get_output()
+
+        self._output = signal
+
+    def get_output(self) -> Field:
+        """Get the generated signal as a DPF field.
+
+        Returns
+        -------
+        Field
+            Generated signal as a DPF field.
+        """
+        if self._output == None:
+            warnings.warn(
+                PyAnsysSoundWarning(
+                    "Output is not processed yet. "
+                    f"Use the {__class__.__name__}.process() method."
+                )
+            )
+        return self._output
+
+    def get_output_as_nparray(self) -> np.ndarray:
+        """Get the generated signal as a NumPy array.
+
+        Returns
+        -------
+        numpy.ndarray
+            Generated signal as a NumPy array.
+        """
+        output = self.get_output()
+
+        if output == None:
+            return np.array([])
+
+        return np.array(output.data)
+
+    def plot(self):
+        """Plot the resulting signal in a figure."""
```

Comment:
@ansaminard here too we should not switch on the source type, but have a more object oriented approach by asking each source to do something. I'm looking at a way to do it smartly with matplotlib objects

Reply:
done (as discussed, track plots the filtered signal, and then calls the source's new plot_control method that plots control(s), if they exist)